gen/types.go

```diff
@@ -55,6 +57,28 @@ func processTypeReplacements[C, I any](types drivers.Types, replacements []Repla
 	}
 }
 
+// matchString reports whether string a matches a pattern.
+// Pattern a can be either a literal string (case-insensitive comparison)
+// or a regular expression enclosed with /. Regex patterns are automatically made case-insensitive.
+func matchString(pattern, candidate string) bool {
+	stringPatterns, regexPatterns := drivers.ClassifyPatterns([]string{pattern})
+	for _, pattern := range stringPatterns {
+		if strings.ToLower(pattern) == strings.ToLower(candidate) {
```

Comment:
```suggestion
		if strings.EqualFold(pattern, candidate) {
```

Reply:
I thought about this but this also allows `ß` and `ss` to match, so I'm not sure.

Reply:
No, I actually misread the docs

https://pkg.go.dev/strings#EqualFold

> `fmt.Println(strings.EqualFold("ß", "ss"))  // false because comparison does not use full case-folding`